hbase-server/src/main/java/org/apache/hadoop/hbase/master/http/hbck/resource/HbckMetricsResource.java

```diff
@@ -0,0 +1,138 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.http.hbck.resource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.client.RegionInfo;
+import org.apache.hadoop.hbase.master.MasterServices;
+import org.apache.hadoop.hbase.master.hbck.HbckReport;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckEmptyRegionInfo;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckInconsistentRegions;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckMetrics;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckOrphanRegionsOnFS;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckOrphanRegionsOnRS;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckOverlapRegions;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckRegionDetails;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckRegionHoles;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckServerName;
+import org.apache.hadoop.hbase.master.http.hbck.model.HbckUnknownServers;
+import org.apache.hadoop.hbase.master.janitor.CatalogJanitorReport;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.yetus.audience.InterfaceAudience;
+
+import org.apache.hbase.thirdparty.javax.ws.rs.GET;
+import org.apache.hbase.thirdparty.javax.ws.rs.Path;
+import org.apache.hbase.thirdparty.javax.ws.rs.Produces;
+import org.apache.hbase.thirdparty.javax.ws.rs.core.MediaType;
+
+@Path("hbck_metrics")
+@Produces({ MediaType.APPLICATION_JSON })
+@InterfaceAudience.Private
+public class HbckMetricsResource {
+  private final HbckReport hbckReport;
+  private final CatalogJanitorReport catalogJanitorReport;
+
+  @Inject
+  public HbckMetricsResource(MasterServices master) {
+    this.hbckReport = master.getHbckChore().getLastReport();
+    this.catalogJanitorReport = master.getCatalogJanitor().getLastReport();
+  }
+
+  @GET
+  public HbckMetrics getBaseHbckMetrics() {
+    return new HbckMetrics(hbckReport.getCheckingStartTimestamp().toEpochMilli(),
+      hbckReport.getCheckingEndTimestamp().toEpochMilli(), getOrphanRegionsOnFS(),
+      getOrphanRegionsOnRS(), getInconsistentRegions(), getRegionChainHoles(),
+      getRegionChainOverlap(), getUnknownServers(), getEmptyRegionInfo());
+  }
+
+  @GET
+  @Path("/orphan_regions_on_fs")
```

Comment:
I read at places using hyphens is a recommended approach in REST APIs - https://restfulapi.net/resource-naming/

Reply:
Done